File: src/main/java/no/novari/flyt/authorization/client/sourceapplications/model/EgrunnervervSourceApplication.java
package no.novari.flyt.authorization.client.sourceapplications.model;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class EgrunnervervSourceApplication extends BaseSourceApplication {
    public EgrunnervervSourceApplication(@Value("${fint.flyt.egrunnerverv.sso.client-id:#{null}}") String clientId) {
        super(2L, "eGrunnerverv", clientId);
    }
}

File: src/main/kotlin/no/novari/flyt/authorization/client/sourceapplications/model/EgrunnervervSourceApplication.kt
package no.novari.flyt.authorization.client.sourceapplications.model

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class EgrunnervervSourceApplication(
    @Value("\${fint.flyt.egrunnerverv.sso.client-id:#{null}}")
    clientId: String?,
    @Value("\${fint.flyt.egrunnerverv.available:true}")
    available: Boolean,
) : BaseSourceApplication(
        id = 2L,
        displayName = "eGrunnerverv",
        clientId = clientId,
        available = available,
    )
